crates/forge/src/multi_runner.rs

```diff
@@ -275,15 +275,21 @@ impl MultiContractRunnerBuilder {
                 } = post_link_input;
                 let dependencies = unique_deps(dependencies);
 
-                // get bytes
-                let bytecode =
-                    if let Some(b) = contract.bytecode.expect("No bytecode").object.into_bytes() {
-                        b
+                let abi = contract.abi.expect("We should have an abi by now");
+
+                // get bytes if deployable, else add to known contracts and return
+                let bytecode = if let Some(bytecode) = contract.bytecode {
+                    if let Some(bytes) = bytecode.object.into_bytes() {
+                        bytes
                     } else {
+                        known_contracts.insert(id.clone(), (abi, vec![]));
                         return Ok(())
-                    };
+                    }
+                } else {
+                    known_contracts.insert(id.clone(), (abi, vec![]));
+                    return Ok(())
+                };
```

Comment:
Can we get the rationale for this change? This code is extremely sensible—it touches how we link contracts and we've had serious issues in the past on this.

Reply:
As mentioned in another comment, currently any abi without bytecode is ignored. So if the contract is not deployable we just add it to `known_contracts`, but not `deployable_contracts`.

Ideally looking for the inverse of the `and_then` chain at the end of the function.